design-docs/2021/q1/plugin-redesign.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Plugin Design Doc
+author: Kevin Siegler (@topocount)
+created: 2021-02-02
+---
+
+# Plugin Design Doc
+
+## Overview
+
+SourceCred plugins are almost completely beholden to their datasources.
+A local cache exists, but its only real purpose is to provide a local source of data for graph creation.
+Plugin caches cannot stand by themselves for too long and are easily blown out by upstream changes.
+Additionally, caches are only built in environments local to graph creation.
+They cannot easily be shared over a network, or provide any real source of truth outside of the API.
+
+@hammadj has designed an new [plugin model] that relies upon a `DataProvider` and `DataArchive` as a persistent mirror of data from a specific platform.
```

Comment:
Should update the terminology here to reflect the latest naming we agreed on, `Fetcher` and `DataProvider` respectively